Fix typos in basics/functions

wilzbach opened the pull request.

wilzbach commented on public/content/en/basics/functions.md:
```diff
@@ -1,17 +1,15 @@
 # Functions
 
 You've seen one function already: `main()` - the starting point of every
```
We use "you" here, but I couldn't find a nice way to say this without ...

stonemaster replied:
- One function has already been introduced:

wilzbach commented on public/content/en/basics/functions.md:
```diff
@@ -20,7 +18,7 @@ aware of that.
 Functions might even be declared inside others functions where they may be
 used locally and aren't visible to the outside world.
 These function can even have access to objects local to
-the parent's scope
```
we introduce delegates here, even though there's an entire section about them:

http://tour.dlang.io/tour/en/basics/delegates

stonemaster replied:
Good idea!

wilzbach replied:
Added "Such nested functions are called delegates and will be explained in more depth soon."

ZombineDev commented on public/content/en/basics/functions.md:
```diff
@@ -1,17 +1,15 @@
-the function's body don't match the compiler will certainly make you
-aware of that.
+If the return type is defined as `auto`, the D compiler infers the return
+type automatically. Hence multiple `return` statements must return the same type.
```
... `return` statements must return **values with compatible types**. See https://dpaste.dzfl.pl/76d3268d5a66 for reference.

wilzbach commented:
> ... return statements must return values with compatible types. See https://dpaste.dzfl.pl/76d3268d5a66 for reference. Though I incidentally found a bug in the compiler.

Wow. I added a small example to depict such a case ;-)
(and also added a link to "soon" for the impatient).

ZombineDev commented on public/content/en/basics/functions.md:
```diff
@@ -1,26 +1,32 @@
-`void` if nothing is returned - and an arbitrary number of parameters.
+One function has already been introduced: `main()` - the starting point of every
+D program. A function may return a value - or be declared with
+`void` if nothing is returned - and accept an arbitrary number of input parameters.
```
I think this should be "arbitrary number of **arguments**".

wilzbach commented:
> I think this should be "arbitrary number of arguments".

(fixed)

If you find the time, you should definitely go through the tour! ;-)

ZombineDev commented:
> (fixed)

Thanks, all LGTM.

> If you find the time, you should definitely go through the tour! ;-)

It's been ib my TODO list for while. Hopefully in a couple of weeks I'll have more free time.
